### PR TITLE
getting last message of recent channels-backend

### LINKED ISSUE #4: Render a list of recent conversations
## User story

**Who:** **As a** user

**What:** **I want to** See the users i latest chatted with

**Why:** **so that we can** quickly chat with them again

## Implementation details
- What if there are no latest users? Maybe a "create your first conversation" button
- Limit the number of conversations to 20
- See figure 1 for ui
- You need to show: profile image, name of person, the latest message (one line) and time of message
- If there is no api, make your own super simple api that just returns 3 users with image, name, etc. Talk to the person building the api. How will the data be structured? You want to have the same structure in your test api as the correct one

nice to have as extra PR: automatic load new conversations when scrolling down

### PR DESCRIPTION
**Pull request by afrouzhakim**
# Description
get the last message and the title of the channels, from the recent channels for a specific user that defined in the endpoint. 

Fixes #4 

# How to test?
run "npm run server" and go to "http://localhost:5000/api/channels-message/2"

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by benna100 — CHANGES_REQUESTED** [review]
Hey Afrouz...

As far as i can see, then what you are doing here can be done in the frontend. If you want message for a specific user just use `/api/messages?sender=3` With the messages you can figure out what channels they belong to and theeby render the recent messages (or am i missing somthing here?)

The story was a frontend task that changes the UserList to recentMessages. I can see that you have spent a long time on this, and its actually pretty impressive, so really good job on getting this to work. But if you had asked about this task or communicated a little, you could have save probably a weeks development. And all this code we dont really need cause the api already supports it. And that sucks that you have made all this (super impressive work) that we dont get to use.

**Comment by afrouzhakim** [comment]
Hi Benjamin, 
thank you for your feedback. Yes you are right I must to have a better overview of my classmates work before start coding! I will try to use the API that you mentioned and complete my task on that.

File: src/server/api/controllers/channels-message-controller.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const knex = require('../../config/db');
+
+const Error = require('../lib/utils/http-error');
+
+// select fk_channel_id from channel_members where fk_user_id=userId;
+
+const getChannelsId = async (userId) => {
+  try {
+    console.log(userId);
```
Comment (tpetrina) [inline src/server/api/controllers/channels-message-controller.js:9]: Remove console log from here.

File: src/server/api/controllers/channels-message-controller.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const channels = await knex('channel_members')
+      .select('fk_channel_id')
+      .where('fk_user_id', '=', userId);
+    console.log(channels);
```
Comment (tpetrina) [inline src/server/api/controllers/channels-message-controller.js:13]: Remove console.log and leave an empty line here.

File: src/server/api/controllers/channels-message-controller.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    if (channels.length === 0) {
+      throw new Error(`create your first conversation!`);
```
Comment (tpetrina) [inline src/server/api/controllers/channels-message-controller.js:15]: Don't throw, return `[]` instead.

File: src/server/api/routes/channels-message-router.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const express = require('express');
+
+const router = express.Router({ mergeParams: true });
+
+const channelsMessageController = require('../controllers/channels-message-controller');
+
+router.get('/:userId', (req, res, next) => {
```
Comment (tpetrina) [inline src/server/api/routes/channels-message-router.js:7]: Missing Swagger docs.

File: src/server/api/controllers/channels-message-controller.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        const message = await knex('channel_messages')
+          .select('message', 'title')
+          .innerJoin(
+            'channels',
+            'channels.id',
+            '=',
+            'channel_messages.fk_channel_id',
+          )
+          .where('fk_channel_id', '=', element.fk_channel_id)
+          .orderBy('channel_messages.updated_at', 'asc')
```
Comment (tpetrina) [inline src/server/api/controllers/channels-message-controller.js:28]: I think you should use `desc` right? Get the latest message so `desc`.

File: src/server/api/controllers/channels-message-controller.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+const knex = require('../../config/db');
+
+const Error = require('../lib/utils/http-error');
+
+// select fk_channel_id from channel_members where fk_user_id=userId;
```
Comment (tpetrina) [inline src/server/api/controllers/channels-message-controller.js:5]: Please remove it.

**Review by tpetrina — APPROVED** [review]
Great job! Please remove a couple of files and lines and that's basically it!.

**Review by benna100 — APPROVED** [review]

**Pull request merged**